soccer/gameplay/skills/goalside_mark.py

```diff
@@ -0,0 +1,194 @@
+import single_robot_behavior
+import behavior
+import robocup
+import main
+import constants
+import role_assignment
+
+class Goalside_Mark(single_robot_behavior.SingleRobotBehavior):
+    '''
+    The goalside mark will take a robot or mark point and position itself on line with the best shot from 
+    the position of the robot/mark_point.
+    @param: ratio - this will control how far from the opposing robot the defender will mark (0 is close to robot)
+    '''
+
+    
+    def __init__(self):
+        super().__init__(continuous=True)
+        #Below params are described above @properties
+        self._ratio = 0.2
+        #This threshold is lower than mark to ensure we get goal side of the opponent fast, 
+        #then back off based on ratio
+        
+        self._mark_line_thresh = constants.Robot.Radius*3
+        self._mark_robot = None 
+        self._mark_point = None 
+
+        self._target_point = None 
+
+        self.add_transition(behavior.Behavior.State.start,
+                            behavior.Behavior.State.running, lambda: True,
+                            "immediately")
+        self.kick_eval = robocup.KickEvaluator(main.system_state())
+
+    def execute_running(self):
+        #pylint: disable=no-member
+        #Skill does nothing if mark point isn't given AND the ball or robot to mark can't be found
+        self.kick_eval = robocup.KickEvaluator(main.system_state())
+        if self.mark_point is None and \
+           (self.mark_robot is None or
+            not main.ball().valid or
+            not self.mark_robot.visible):
+            return
+
+        pos = self.robot.pos
+        mark_pos = self.mark_point if self.mark_point is not None else self.mark_robot.pos
+        self.kick_eval.add_excluded_robot(self.robot)
+
+        shot_pt, shot_score = self.kick_eval.eval_pt_to_our_goal(mark_pos)
+        self.kick_eval.excluded_robots.clear()
+
+        
+        #Finds the line from the mark position to the shot point and creates a line between them
+        #removing the overlap with the ball on one side and robot on the other
+        #This assumes even with mark position parameter that there is a robot there to avoid
+        mark_line_dir = (mark_pos - shot_pt).normalized()
+        mark_line, shot_pt = self.get_mark_line()
+        print(mark_line)
```

Comment:
In the final version, prints should probably be removed so we don't spam the console too much